lib/tasks/errors.rake

```diff
@@ -15,39 +15,57 @@ namespace :errors do
 
   task :fetch_log => :environment do
     request_id = ENV['request_id']
-    logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
     unless request_id
       puts "Can't find log without request_id"
       exit(1)
     end
-    type = Foreman::Logging.config[:type]
-    unless type == 'file'
-      puts "This rake task only support log file, you're using #{type}"
-      puts "You can search the logs in #{type} for request_id #{request_id}"
-      exit(1)
-    end
-    if Foreman::Logging.config[:layout] != 'multiline_request_pattern'
-      puts "Warning: Logging layout is not multiline_request_pattern."
-      puts "This output of this command can be incomplete."
-    end
-    file_path = File.join(Foreman::Logging.log_directory, logfile)
-    unless File.exist?(file_path)
-      puts "Can't find log file #{file_path}"
-      exit(1)
-    end
-    result = `grep "#{request_id}" "#{file_path}"`
-    if result.empty?
-      puts "Can't find log for request_id"
-      exit(1)
-    end
-    puts result
-    puts "\n"
+
     puts "Foreman version: #{Foreman::Version.new.full}"
     unless (plugins = Foreman::Plugin.all).empty?
       puts "Plugins: "
       plugins.each do |plugin|
         puts " - #{plugin.name} #{plugin.version}"
       end
     end
+    puts
+
+    type = Foreman::Logging.config[:type]
+    lines = case type
+            when 'file'
+              logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
```

Comment:
I just realized we never did this and probably out of scope for this PR, but we can extract the filename from the logger config.